src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +24,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle current = (Rectangle) rectangle;
+            return Objects.equals(this.width, current.width)
+                  && Objects.equals(this.length, current.length)
+                  && Objects.equals(this.color, current.color);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int hashCode = 17;
+        hashCode = 31 * hashCode + (width != null ? width.hashCode() : 0)
```

Comment:
it's better to repeat the exactly same operation for each field

Reply:
hashCode = 31 * hashCode + (width != null ? width.hashCode() : 0);
hashCode = 31 * hashCode + (length != null ? length.hashCode() : 0);
//and so on